src/main/java/com/kelinci/mainapp/api/UserController.java

```diff
@@ -26,8 +29,13 @@ public OurUser getOurUsers() {
 
     @GetMapping(value = "/registered/listofusers")
     //czeka na wywolanie localhost:8080/users
-    public String getListOfRegisteredUsers() {
-        return listOfUsers.toString();
+    public List<RegisteredUserResponse> getRegisteredUserResponse() {
+
+        //mapujemy tutaj ourUser na ResitsterUR - chodzi o to żeby ujawnić tylko to co ma być zdefiniowane w RUR
+
+        return listOfUsers.stream()
+                .map(ourUser -> new RegisteredUserResponse(ourUser.getMail()))
```

Comment:
Implementacje można schować do prywatnej metody
private RegisteredUserResponse toRegisteredUserResponse(OurUser ourUser) {
...
}

a następnie przeniesiemy tą prywatną metodę do jakiejś specjalnej klasy -> Mappera :O

Reply:
tutaj nie rozumiem czemu chcemy przeniesc do specjalnej klasy - może call jutro pliz

Reply:
Bo Mapper chyba musi mieć najpierw dostęp do listOfUsers - a w tym momencie nie będzie miał bo to pole private w UserController

Reply:
Chcemy zrobić zmianę w dwóch krokach:
1. chcemy wrzucić implementację do prywatnej metody. Bo ja nie chcę tutaj widzieć JAK mapujesz jeden obiekt w drugi tylko chcę widzieć CO robisz (czyli że mapujesz). Więc jakbym widział w linii 37 takie coś:
`.map(ourUser -> toRegisteredUserResponse(ourUser))`

a po refaktorze IntellJowym takie coś:
`.map(this::toRegisteredUserResponse(ourUser))`

ot byłoby super czytelne.

2. w drugim kroku uświadomimy sobie, że dlaczego UserController musi umieć (ma wszak prywatną metodę) mapować jeden obiekt na drugi? Niech to robi dedykowany mapper